File: cpp/p030.cc
# include <iostream>

int fifth_power_digit_sum(int x);

int compute();


int main(){

	using std::cout;
	using std::endl;

	cout<<compute()<<endl;

}


int fifth_power_digit_sum(int x){
	int ans=0;

	while(x>=1){
		int m=x%10;
		ans+=m*m*m*m*m;
		x/=10;
	}
	return ans;
}


int compute(){
	int ans=0;
    const int N=1000000;
	for(int i=2;i<=N;i++){
		if( i==fifth_power_digit_sum(i)){
			ans+=i;
		}

	}

	return ans;
}

File: julia/p030.jl

function fifth_power_digit_sum(n::Int)

	a=Int[];

	while(n>=1)
		push!(a,n%10);
		n=n-n%10;
		n/=10;
		n=Int(n)
	end
	ans=0;
	for i in a
		ans+=i^5
	end
	return ans
end


function compute()
	ans=0
	for i=2:1000000
		if i==fifth_power_digit_sum(i)
			ans+=i
		end
	end
	return ans
end

println(compute())